src/App.jsx

```diff
@@ -1,248 +1,362 @@
-              className="button mr-2 my-1"
-              href="#/"
-            >
-              Category 2
-            </a>
-
-            <a
-              data-cy="Category"
-              className="button mr-2 my-1 is-info"
-              href="#/"
-            >
-              Category 3
-            </a>
-            <a
-              data-cy="Category"
-              className="button mr-2 my-1"
-              href="#/"
-            >
-              Category 4
-            </a>
-          </div>
-
-          <div className="panel-block">
-            <a
-              data-cy="ResetAllButton"
-              href="#/"
-              className="button is-link is-outlined is-fullwidth"
-            >
-              Reset all filters
-            </a>
-          </div>
-        </nav>
-      </div>
-
-      <div className="box table-container">
-        <p data-cy="NoMatchingMessage">
-          No products matching selected criteria
-        </p>
-
-        <table
-          data-cy="ProductTable"
-          className="table is-striped is-narrow is-fullwidth"
-        >
-          <thead>
-            <tr>
-              <th>
-                <span className="is-flex is-flex-wrap-nowrap">
-                  ID
-
-                  <a href="#/">
-                    <span className="icon">
-                      <i data-cy="SortIcon" className="fas fa-sort" />
-                    </span>
+import usersFromServer from './api/users';
+import categoriesFromServer from './api/categories';
+import productsFromServer from './api/products';
+
+const products = productsFromServer.map((product) => {
+  const category = categoriesFromServer
+    .find(categoryData => categoryData.id === product.categoryId);
+
+  const user = usersFromServer
+    .find(userData => userData.id === category.ownerId);
+
+  return {
+    ...product,
+    category,
+    user,
+  };
+});
+
+const users = [{ id: 0, name: 'All' }, ...usersFromServer];
+
+const getPreparedProducts = (
+  productsData,
+  owner,
+  query,
+  selectedCategories,
+) => {
+  let resultData = [...productsData];
+
+  if (owner) {
+    if (owner === 'All') {
+      resultData = [...products];
+    } else {
+      resultData = resultData.filter(({ user }) => user.name === owner);
+    }
+  }
+
+  if (query) {
+    resultData = resultData
+      .filter(({ name }) => name.toLowerCase().startsWith(query.toLowerCase()));
+  }
+
+  if (selectedCategories.length > 0) {
+    if (selectedCategories[selectedCategories.length - 1] === 'All') {
+      if (owner && owner === 'All') {
+        resultData = [...products];
+      } else {
+        resultData = [...resultData];
+      }
+    } else {
+      resultData = resultData
+        .filter(({ category }) => selectedCategories.includes(category.title));
+    }
+  }
+
+  return resultData;
+};
+
+export const App = () => {
+  const [filterByOwner, setFilterByOwner] = useState('All');
+  const [filterByCategories, setFilterByCategories] = useState(['All']);
```

Comment:
```suggestion
  const [filterByCategories, setFilterByCategories] = useState([]);
```
You can define the state by default like an empty array instead of `All` in the array